app/src/main/java/com/androidavanzado/covid19stadistics/ui/DatePickerFragment.kt

```diff
@@ -0,0 +1,28 @@
+package com.androidavanzado.covid19stadistics.ui
+
+import android.app.DatePickerDialog
+import android.app.Dialog
+import android.content.Context
+import android.os.Bundle
+import android.widget.DatePicker
+import androidx.fragment.app.DialogFragment
+import java.util.Calendar
+
+class DatePickerFragment (val listener: (day : Int, month: Int, year: Int) -> Unit) :
+        DialogFragment(), DatePickerDialog.OnDateSetListener {
+
+    override fun onDateSet(view: DatePicker?, year: Int, month: Int, dayOfMonth: Int) {
+
+        val newMonth = month + 1
+        listener(dayOfMonth, newMonth, year)
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+
+        val c = Calendar.getInstance()
```

Comment:
change this variable name to another one more verbose like "calendar"

Reply:
done